src/charm.py

```diff
@@ -930,6 +979,21 @@ def _generate_middleware_config(
           "cannot create middleware: multi-types middleware not supported, consider declaring two
           different pieces of middleware instead"
         """
+        forwardauth_middleware = {}
+        if self.config["enable_experimental_forward_auth"]:
+            if self.forward_auth.is_ready():
+                forward_auth_config = self.forward_auth.get_provider_info()
+                if data.get("name") in forward_auth_config.app_names:  # type: ignore
+                    policy_decision_point_app = self.forward_auth.get_remote_app_name()
+                    forwardauth_middleware[
+                        f"juju-sidecar-forward-auth-{policy_decision_point_app}"
```

Comment:
I think we need to suffix the middleware with the `prefix`
```suggestion
                        f"juju-sidecar-forward-auth-{prefix}"
```